Automattic/jetpack
# JITM: Prevent JITMs from being registered more than once

#### Changes proposed in this Pull Request:
If multiple consumer plugins use the JITM package, it may be possible to register JITMs multiple times, resulting in unnecessary API requests. To prevent this, prevent the JITMs from being registered more than once.

* Add a new class property, `JITM::jitms_registered`, which is used to determine whether the JITMs have already been registered.
* Check the value of `JITM::jitms_registered` before registering the JITMs.
* Add a new method, `JITM::jitms_enabled`, which is used to determine whether JITMs are enabled.
* Update the unit tests.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:

To demonstrate that these changes prevent multiple requests for the JITMs, we'll first cause multiple requests to be generated, then apply this branch to confirm that only one request is generated.

##### Cause duplicate JITM requests
1. Install and activate the master branch of Jetpack.
2. Connect Jetpack.
3. Add this code in a snippet (you can use something like the Code Snippets plugin if you'd like):
    `( new \Automattic\Jetpack\JITMS\JITM() )->configure();`
4. Open your browsers inspector.
5. Navigate to an admin page and search the network requests for `jitm`. You should see two requests to the `wp-json/jetpack/v4/jitm` endpoint.

##### Test this branch
6. Apply this branch.
7. Navigate to an admin page and search the network requests for `jitm`. You should see one request to the `wp-json/jetpack/v4/jitm` endpoint.
